modules/publication/ajax/FileUpload.php

```diff
@@ -0,0 +1,877 @@
+    }
+}
+
+/**
+ * Insert new collaborators into collaborator table and/or put them in rel table
+ * Does NOT include insert for lead investigator
+ *
+ * @param int $pubID the publication ID
+ *
+ * @return void
+ */
+function insertCollaborators(int $pubID) : void
+{
+    if (!$_POST['collaborators']) {
+        return;
+    }
+    $db = Database::singleton();
+
+    $collaborators = json_decode($_POST['collaborators'], true);
+    foreach ($collaborators as $c) {
+        $cid = $db->pselectOne(
+            'SELECT PublicationCollaboratorID '.
+            'FROM publication_collaborator '.
+            'WHERE Name=:c',
+            array('c' => $c['name'])
+        );
+        // if collaborator does not already exist in table, add them
+        if (!$cid) {
+            $collabInsert = array(
+                             'Name'  => $c['name'],
+                             'Email' => $c['email'],
+                            );
+
+            $db->insert(
+                'publication_collaborator',
+                $collabInsert
+            );
+            $cid = $db->pselectOne(
+                'SELECT PublicationCollaboratorID ' .
+                'FROM publication_collaborator ' .
+                'WHERE Name=:c',
+                array('c' => $c['name'])
+            );
+        }
+        $collabRelInsert = array(
+                            'PublicationID'             => $pubID,
+                            'PublicationCollaboratorID' => $cid,
+                           );
+        $db->insertIgnore(
+            'publication_collaborator_rel',
+            $collabRelInsert
+        );
+    }
+}
+
+/**
+ * Inserts users with edit access for the given publication project
+ *
+ * @param int $pubID publication ID
+ *
+ * @return void
+ */
+function insertEditors(int $pubID) : void
+{
+    if (empty($_POST['usersWithEditPerm'])) {
+        return;
+    }
+
+    $db = Database::singleton();
+    $usersWithEditPerm = json_decode($_POST['usersWithEditPerm']);
+    foreach ($usersWithEditPerm as $uid) {
+        $insert = array(
+                   'PublicationID' => $pubID,
+                   'UserID'        => $uid,
+                  );
+
+        $db->insertIgnore(
+            'publication_users_edit_perm_rel',
+            $insert
+        );
+    }
+}
+
+/**
+ * Inserts keywords for the given publication project
+ *
+ * @param int $pubID publication ID
+ *
+ * @return void
+ */
+function insertKeywords(int $pubID) : void
+{
+    if (empty($_POST['keywords'])) {
+        return;
+    }
+    $db       = Database::singleton();
+    $keywords = json_decode($_POST['keywords']);
+    foreach ($keywords as $kw) {
+        // check if keyword exists
+        $kwID = $db->pselectOne(
+            'SELECT PublicationKeywordID ' .
+            'FROM publication_keyword ' .
+            'WHERE Label=:kw',
+            array('kw' => $kw)
+        );
+        // if it doesn't, add it to keyword table and retrieve ID
+        if (!$kwID) {
```

Comment:
can you make this a more precise check like `$kwID !== null` or `!empty($kwID)`?